### PR TITLE
implement `hdc.algo.ops.stats`

### PR DESCRIPTION
**Pull request by valpesendorfer**
This PR

- adds tooling to calculate Mann-Kendall trend metrics 
- wraps everything together with the `spi` functions into a new `stats` module

### CODE REVIEW

File: hdc/algo/ops/stats.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    n = len(x)
+    _s1 = 0
+    _s2 = 0
+
+    for k in range(0, n - 1):
+        for kk in range(k + 1, n):
+            if x[kk] > x[k]:
+                _s1 += 1
+            if x[kk] < x[k]:
+                _s2 += 1
```
Comment (trenouard) [inline hdc/algo/ops/stats.py:229]: Is the double loop more efficient in nopython jit than the numpy operations in the original pymankendall?
```
for k in range(n-1):
        s = s + np.sum(demo[k+1:n][x[k+1:n] > x[k]]) - np.sum(demo[k+1:n][x[k+1:n] < x[k]])
```        
If so, that's good to know!

File: hdc/algo/ops/stats.py [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    """Calculate Mann-Kendall's p value and significance.
+
+    https://vsp.pnnl.gov/help/vsample/design_trend_mann_kendall.htm
+
+    Hussain et al., (2019). pyMannKendall:
+    a python package for non parametric Mann Kendall family of trend tests..
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    """
+    p = 2 * (1 - (0.5 * (1.0 + erf(abs(z) * sqrt(0.5)))))
+    h = int(abs(z) > sc.ndtri(1 - alpha / 2))
```
Comment (trenouard) [inline hdc/algo/ops/stats.py:294]: Do you have a link to scipy code or from where you got the formula for the `cdf` and `ppf` here? That's super nice!

**Review by trenouard — APPROVED** [review]
Thanks for integrating this @valpesendorfer, it looks great! The modular implementation will make it easier if we also want to try out other mann kendall tests (like the seasonal one), so nice

File: hdc/algo/ops/stats.py (hunk 1 above)
Comment (valpesendorfer) [reply hdc/algo/ops/stats.py:229]: yes in general it's more efficient - although good to check on a by case basis

File: hdc/algo/ops/stats.py (hunk 2 above)
Comment (valpesendorfer) [reply hdc/algo/ops/stats.py:294]: see 

https://stats.stackexchange.com/questions/187828/how-are-the-error-function-and-standard-normal-distribution-function-related 

and 

https://stackoverflow.com/questions/20626994/how-to-calculate-the-inverse-of-the-normal-cumulative-distribution-function-in-p

**Review by paololucchino — APPROVED** [review]
Thanks @valpesendorfer !

**Pull request merged**